src/validation.cpp

```diff
@@ -484,6 +486,7 @@ int GetUTXOConfirmations(const COutPoint& outpoint)
 
 bool CheckTransaction(const CTransaction& tx, CValidationState &state)
 {
+    boost::posix_time::ptime start = boost::posix_time::microsec_clock::local_time();
```

Comment:
Depending on how close we want to stick to the statstoshi implementation we should consider using `cxxtimer::Timer` for such measurements as they cause less noise in the code.

Reply:
Also, when it comes to measuring time spent in methods, it might make sense to introduce some RAII type objects which report the time on destructor call.